### ISSUE TITLE
Moving more logic to the Auth service ?

### ISSUE THREAD

**Issue by wladston** [fnakstad/angular-client-side-auth#8]
Hello,

I'm implementing something based in your code, thanks :)

I was thinking, maybe it would be a good idea to pass the rootScope manipulation logic for inside the Auth service ? Whenever a login or a logout is made, we _always_ need to set these variables.

What do you think ?

**Comment by fnakstad** [comment]
Yep! You read my mind :) I'm actually working on a bigger refactoring of both the server and client side code, part of which encompasses putting more of the authentication/authorization specific stuff into the Auth service. So expect a bigger update before the week is over!

**Comment by wladston** [comment]
Awesome! I'll watch closely. I'll also try to make my own version of the Auth service while you finish yours.

**Comment by fnakstad** [comment]
Cool! Definitely let me know if you have any other issues or ideas to make the solution better :)

**Comment by wladston** [comment]
I do have a few, but I'll wait for you to release your updated version, maybe you already tackled them. As soon as I commit my implementation I'll send it to you as well (I'm doing a Bulletin Board).

**Comment by fnakstad** [comment]
Okay, I just pushed whatever I have right now though there will be more refactoring once I've cleared off some more pressing issues.

**Comment by wladston** [comment]
Awesome. Will check it out tonight.

**Comment by wladston** [comment]
One question: why do you get the cookie with the user name twice in the service.js ? In my local tests, the call in init is unnecessary ...

**Comment by wladston** [comment]
Here is my current implementation (considerably simpler than yours): https://github.com/wladston/mongobb

**Comment by fnakstad** [comment]
Ah, the init function isn't actually called anywhere. I simply forgot to remove it, but it's gone now :) It looks like we've done the exact same change in the `login`, `logout`, and `register` functions in the Auth service, so that's a good sign. I also moved some of the setup logic I used to have in my Angular `run()` handler to the Auth service instead, and added some functions (`authorize`, `isLoggedIn`) which I am using in the `$routeChangeStart` handler.
